packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -202,12 +203,44 @@ describe('<Autocomplete />', () => {
           renderInput={(params) => <TextField {...params} autoFocus />}
         />,
       );
-
       checkHighlightIs(getByRole('listbox'), 'one');
       setProps({ value: 'two' });
       checkHighlightIs(getByRole('listbox'), 'two');
     });
 
+    it('should have focused on first element after navigating all elements once by keyboard', () => {
+      const { getByRole } = render(
+        <Autocomplete
+          open
+          options={['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten']}
+          renderInput={(params) => <TextField {...params} />}
+          ListboxProps={{ style: { padding: 0 } }}
```

Comment:
Let's explicitly restrict the height of the listbox by setting `maxHeight: '100px`. Also such a lower value could let us test with fewer options (5 should be enough).